pkg/controllers/managementlegacy/globaldns/globaldnsprovider_catalog_launcher.go

```diff
@@ -276,17 +266,37 @@ func (n *ProviderCatalogLauncher) createUpdateExternalDNSApp(obj *v3.GlobalDnsPr
 	return nil, nil
 }
 
-func (n *ProviderCatalogLauncher) getProviderIfAlreadyRunning(obj *v3.GlobalDnsProvider) (*pv3.App, error) {
+func (n *ProviderCatalogLauncher) deleteExternalDNSApp(key string, obj *v3.GlobalDnsProvider) (runtime.Object, error) {
+	var err error
+	_, globalDNSproviderName, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return obj, err
+	}
+	//check if provider already running for this GlobalDNSProvider.
+	existingApp, err := n.getProviderIfAlreadyRunning(globalDNSproviderName)
+	if err != nil {
+		return nil, err
```

Comment:
Do we want to `return obj, err` instead here? Is the idea that we return `nil` if deletion was successful and `obj` otherwise? If that's not the case, we could simply remove `obj` parameter from this func as we don't need it here (and it's quite likely `nil` anyway) and we could change calling function to `return obj, n.deleteExternalDNSApp(key)`

Reply:
done, will return ``` return obj, nil ```

Reply:
update: we should return nil, nil in this case since there is no update.